uint32           cam_idx
string           cam_geom
float64[2]       resolution
float64[]        params
